File: Examples/CSharp/ProgrammersGuide/Slides-Presentation/ThumbnailWithUserDefinedDimensions.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2013 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Slides. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Slides;
using System.Drawing;

namespace CSharp.Slides
{
    public class ThumbnailWithUserDefinedDimensions
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Slides_Presentations();

           //Instantiate a Presentation class that represents the presentation file
            using(Presentation pres = new Presentation(dataDir+ "ThumbnailWithUserDefinedDimensions.pptx"))
{

           //Access the first slide
            ISlide sld = pres.Slides[0];

           //User defined dimension
            int desiredX = 1200;
            int desiredY = 800;

        //Getting scaled value  of X and Y
            float ScaleX = (float)(1.0 / pres.SlideSize.Size.Width) * desiredX;
            float ScaleY = (float)(1.0 / pres.SlideSize.Size.Height) * desiredY;


       //Create a full scale image
            Bitmap bmp = sld.GetThumbnail(ScaleX, ScaleY);

        //Save the image to disk in JPEG format
 bmp.Save(dataDir+ "Thumbnail2.jpg", System.Drawing.Imaging.ImageFormat.Jpeg);

}
            }
            
            
        }
 }

File: Examples/VisualBasic/ProgrammersGuide/Slides-Presentation/ThumbnailWithUserDefinedDimensions.vb
'////////////////////////////////////////////////////////////////////////
' Copyright 2001-2015 Aspose Pty Ltd. All Rights Reserved.
'
' This file is part of Aspose.Slides. The source code in this file
' is only intended as a supplement to the documentation, and is provided
' "as is", without warranty of any kind, either expressed or implied.
'////////////////////////////////////////////////////////////////////////

Imports Microsoft.VisualBasic
Imports System.IO
Imports System.Drawing
Imports Aspose.Slides

'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Slides for .NET API reference 
'when the project is build. Please check https://docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Slides for .NET API from http://www.aspose.com/downloads, 
'install it and then add its reference to this project. For any issues, questions or suggestions 
'please feel free to contact us using http://www.aspose.com/community/forums/default.aspx

Namespace VisualBasic.Slides
    Public Class ThumbnailWithUserDefinedDimensions
        Public Shared Sub Run()
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_Slides_Presentations()

            'Instantiate a Presentation class that represents the presentation file
            Using pres As New Presentation(dataDir & "ThumbnailWithUserDefinedDimensions.pptx")

                'Access the first slide
                Dim sld As ISlide = pres.Slides(0)

                'User defined dimension
                Dim desiredX As Integer = 1200
                Dim desiredY As Integer = 800

                'Getting scaled value  of X and Y
                Dim ScaleX As Single = CSng(1.0 / pres.SlideSize.Size.Width) * desiredX
                Dim ScaleY As Single = CSng(1.0 / pres.SlideSize.Size.Height) * desiredY


                'Create a full scale image
                Dim bmp As Bitmap = sld.GetThumbnail(ScaleX, ScaleY)

                'Save the image to disk in JPEG format
                bmp.Save(dataDir & "Thumbnail2.jpg", System.Drawing.Imaging.ImageFormat.Jpeg)

            End Using
        End Sub


    End Class
End Namespace

